corehq/messaging/smsbackends/turn/models.py

```diff
@@ -39,14 +43,33 @@ def send(self, msg, orig_phone_number=None, *args, **kwargs):
         try:
             wa_id = client.contacts.get_whatsapp_id(to)
         except WhatsAppContactNotFound:
-            pass                # TODO: Fallback to SMS?
+            pass  # TODO: Fallback to SMS?
 
         try:
             message = client.messages.send_text(wa_id, msg.text)
-        except:                 # TODO: Add message exceptions to package
+        except:  # TODO: Add message exceptions to package
             raise
 
     def get_all_templates(self):
         config = self.config
         client = TurnBusinessManagementClient(config.business_id, config.business_auth_token)
         return client.message_templates.get_message_templates()
+
+
+def is_whatsapp_template_message(message_text):
+    return message_text.lower().startswith(WA_TEMPLATE_STRING)
+
+
+def get_template_hsm_parts(message_text):
+    HsmParts = namedtuple("hsm_parts", "template_name lang_code variables")
+    parts = message_text.split("~")[0].split(":")
+
+    try:
+        variables = [p.strip() for p in parts[3].split(",")]
```

Comment:
Are variables guaranteed to not contain commas?

Reply:
Updated here: https://github.com/dimagi/commcare-hq/pull/27193/commits/dfdd1cb5d87c4be02129f3c2b7aeb5b183a5d07d

Thanks!